scripts/publish-npm-packages.sh

```
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+# Set default values for dist-tag and registry for local development
+# to prevent running this script and accidentally publishing to npm
+dist_tag="dev"
+registry="http://localhost:4873"
+
+# shellcheck source=./scripts/helpers/exit-if-fail.sh
+source "$(dirname "$0")/helpers/exit-if-fail.sh"
+
+# check if there were any changes to packages between current and previous commit
+count=$(git diff HEAD~1..HEAD --name-only -- packages | awk '{c++} END {print c}')
+
+# Parse command line arguments
+while [[ $# -gt 0 ]]; do
+    key="$1"
+    case $key in
+        --dist-tag)
+        dist_tag="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        --registry)
+        registry="$2"
+        shift # past argument
+        shift # past value
+        ;;
+        *)    # unknown option
+        echo "Unknown option: $1"
+        exit 1
+        ;;
+    esac
+done
+
+if [ -z "$NPM_TOKEN" ]; then
```

this can be moved up to fail faster

---

+1